Implement: Umbrella Linking

dheidemann opened the pull request:
backend is now implemented @Plebysnacc

Supporting events can be added by doing a `linkSupportingEventToEvent` mutation. To query for events and include all supporting events, add the `includeSupportingEvents: true` argument to the `events` query

backend:
- [x] create db relation "Subscription"
- [x] implement the gql endpoint

frontend:
- [x] in admin edit, add a simple dialog to subscribe other programs
- [x] event dialog: handle reference to the original program

closes https://github.com/FachschaftMathPhysInfo/pepp/issues/180

Plebysnacc commented:
@dheidemann can we have array support for the linking query? Currently I have to link every selected event one by one :c

dheidemann commented:
you can now give `supportingEventID` a slice :3

Plebysnacc commented:
Snegge 🐌

Plebysnacc commented:
so uh... the linking works great... but it fucks with the planner view xd

to reproduce:
- link events
- go to planner :)

Plebysnacc commented:
thx for the addition 🐌

Plebysnacc commented:
That merge conflict was a damn beast...

Plebysnacc commented on server/graph/schema.resolvers.go:
```diff
@@ -776,11 +776,16 @@ func (r *mutationResolver) LinkSupportingEventToEvent(ctx context.Context, event
-func (r *mutationResolver) UnlinkSupportingEventFromEvent(ctx context.Context, eventID int, supportingEventID int) (int, error) {
-	link := &models.EventToSupportingEvent{EventID: int32(eventID), SupportingEventID: int32(supportingEventID)}
+func (r *mutationResolver) UnlinkSupportingEventFromEvent(ctx context.Context, eventID int, supportingEventID []int) (int, error) {
+	var links []int32
+	for _, id := range supportingEventID {
+		links = append(links, int32(id))
+	}
```
Ohh okay, I got a warning in the return types from go in the other project so I did it without testing not using it first

dheidemann commented:
are you okay with me replacing the linking popover with something like this?
![image](https://github.com/user-attachments/assets/3f187213-932b-490b-9d0b-3d1cb1fd3693)

Plebysnacc commented:
I can also implement this, seeing that you've got a lot to do rn

Plebysnacc commented:
I think this is a working version now!

dheidemann commented:
i added the refetch via provider such that after subscribing events get refetched. excluded the source umbrella event from the selection such that you cannot subscribe to yourself. also removed the padding in the popover, hope that is okay for you:
![image](https://github.com/user-attachments/assets/6f4dd189-fde2-4982-86de-8a71ef07c9d7)

from my end this is good to go now, what do you think?

dheidemann commented:
btw, the popover with your implementation looks very nice here, thanks for doing this
